### PR TITLE
Add zero rows to report when needed

### PR DESCRIPTION
**Pull request by rkondratowicz**
Just using the helper from https://github.com/hmcts/bulk-scan-processor/pull/426

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java [hunk 1]
```diff
@@ -15,22 +16,32 @@
     public static final String TEST_JURISDICTION = "BULKSCAN";
 
     private final EnvelopeCountSummaryRepository repo;
+    private final ZeroRowFiller zeroRowFiller;
 
-    public ReportsService(EnvelopeCountSummaryRepository repo) {
+    // region constructor
+    public ReportsService(
+        EnvelopeCountSummaryRepository repo,
+        ZeroRowFiller zeroRowFiller
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java]: Sad it is not utility class

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java [hunk 2]
```diff
@@ -15,22 +16,32 @@
...
         this.repo = repo;
+        this.zeroRowFiller = zeroRowFiller;
     }
+    // endregion
 
     public List<EnvelopeCountSummary> getCountFor(LocalDate date, boolean includeTestJurisdiction) {
-        return repo
-            .getReportFor(date)
+        return zeroRowFiller
+            .fill(repo.getReportFor(date).stream().map(this::fromDb).collect(toList()), date)
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java]: Pushy.. How about reducing unnecessary extra stream?

```java
return zeroRowFiller.fill(
    repo.getReportFor(date)
        .stream()
        .filter(it -> includeTestJurisdiction || !Objects.equals(it.getJurisdiction(), TEST_JURISDICTION))
        .map(this::fromDb)
        .collect(toList()),
    date
);
```

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsServiceTest.java [hunk 3]
```diff
@@ -14,20 +14,24 @@
...
+import static org.mockito.Mockito.when;
 import static uk.gov.hmcts.reform.bulkscanprocessor.services.reports.ReportsService.TEST_JURISDICTION;
 
 @RunWith(MockitoJUnitRunner.class)
 public class ReportsServiceTest {
 
-    @Mock
-    private EnvelopeCountSummaryRepository repo;
+    @Mock private EnvelopeCountSummaryRepository repo;
+    @Mock private ZeroRowFiller zeroRowFiller;
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsServiceTest.java]: As mentioned many many PRs ago - good candidate of not being a mock. But nvm

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java]: filtering need to happen *after* filler does its' thing

**Pull request merged**